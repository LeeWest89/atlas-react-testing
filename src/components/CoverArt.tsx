import placeHolder from '../assets/placeholder.svg';

interface ArtProp {
  artUrl?: string;
}

export default function CoverArt ({ artUrl }: ArtProp) {
  return (
    <div className='flex max-h-[39.9rem] md:h-[25rem] max-w-[39.9rem] md:max-w-[25rem] mb-6'>
      <img src={artUrl || placeHolder} alt="Song Art" />
    </div>
  );
}