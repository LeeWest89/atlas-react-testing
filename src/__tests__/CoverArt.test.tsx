import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import CoverArt from "../components/CoverArt";

// Test artUrl with api
test("CoverArt renders correctly with api", () => {
  const { container } = render(<CoverArt artUrl="https://raw.githubusercontent.com/atlas-jswank/atlas-music-player-api/main/images/albumn4.webp" />);
  expect(container).toMatchSnapshot();
});

// Test artUrl without api
test("CoverArt renders correctly without api", () => {
  const { container } = render(<CoverArt />);
  expect(container).toMatchSnapshot();
});