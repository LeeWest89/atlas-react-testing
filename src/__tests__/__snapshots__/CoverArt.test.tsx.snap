// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt renders correctly with api 1`] = `
<div>
  <div
    class="flex max-h-[39.9rem] md:h-[25rem] max-w-[39.9rem] md:max-w-[25rem] mb-6"
  >
    <img
      alt="Song Art"
      src="https://raw.githubusercontent.com/atlas-jswank/atlas-music-player-api/main/images/albumn4.webp"
    />
  </div>
</div>
`;

exports[`CoverArt renders correctly without api 1`] = `
<div>
  <div
    class="flex max-h-[39.9rem] md:h-[25rem] max-w-[39.9rem] md:max-w-[25rem] mb-6"
  >
    <img
      alt="Song Art"
      src="/src/assets/placeholder.svg"
    />
  </div>
</div>
`;
